gradio/interface.py

```diff
@@ -121,6 +123,18 @@ def get_output_instance(iface):
             except (ImportError, AttributeError):  # If they are using TF >= 2.0 or don't have TF, just ignore this.
                 pass
 
+        if self.allow_flagging:
```

Comment:
I think it's more elegant to come up with the directory name in 1 step rather than 2. So something like:
```
base_dir_name = "-".join(self.title.split(" "))
index = 0
while os.path.exists(os.path.join(FLAGGING_DIRECTORY, base_dir_name , index)):
    index += 1
dir_name = os.path.join(FLAGGING_DIRECTORY, base_dir_name , index)
```

Reply:
To be clear `join` will add a `/` separator so it would be `"FLAGGING_DIRECTORY/base_dir_name/index"`. I thought it would be too nested so I wrote similar code that produces `"FLAGGING_DIRECTORY/base_dir_name_index"`

Reply:
Or do you think nested is better?

Reply:
Ah good point, no I think what you wrote is better